gossip/hooks.py

```diff
@@ -36,14 +36,34 @@ def __init__(self, group, name, arg_names=None, doc=None, deprecated=False, can_
         registry.hooks[self.full_name] = self
         self._registrations = []
         self._empty_regisrations = []
-        self._arguments = self._normalize_arguments(arg_names)
+        self._arguments = None
         self._trigger_internal_hooks = self.full_name != "gossip.on_handler_exception"
         self._defined = False
         self._pre_trigger_callbacks = []
         self._unmet_deps = frozenset()
-        self._can_be_muted = can_be_muted
-        self.doc = doc
-        self.deprecated = deprecated
+        self._can_be_muted = None
+        self.doc = None
+        self.deprecated = None
+        self.set_hook_kwargs(arg_names=arg_names, doc=doc, deprecated=deprecated, can_be_muted=can_be_muted)
+
+    def set_hook_kwargs(self, **kwargs):
+        normalized_args = self._normalize_arguments(kwargs.pop('arg_names', None))
```

Comment:
No check is made for garbage keyword arguments

Reply:
I would rename this function. Perhaps initialize or something like that

Reply:
Perhaps configure?

Reply:
Right... missed the check.
I pushed again, please review.